### PR TITLE
Update contract types and rpc

### PR DESCRIPTION
**Pull request by kwingram25**
Subset of #4423, only updating contract type definitions and RPC call

Not sure what work needs to accompany this for apps. The instantiate request struct fails on older Substrate chains as it doesn't expect storageDepositLimit.

### CODE REVIEW

**Comment by kwingram25** [comment]
@jacogr Noticed that in some cases the rpc for contracts.instantiate returns an ExitError (ie OutOfGas), how would this be represented in the definitions?

**Comment by jacogr** [comment]
Umpf... that is problematic, since generally the RPCs should not break, i.e. they are always backwards compatible. 

I guess in this case it should be up to the users to specify the correct (older) types.

**Comment by kwingram25** [comment]
Anything more to do on our end for this?

File: packages/types/src/interfaces/contracts/definitions.ts [hunk 1]
```diff
@@ -249,22 +255,35 @@ export default {
...
+      result: 'InstantiateReturnValue'
+    },
     InstantiateReturnValueTo267: {
       result: 'ExecReturnValue',
       accountId: 'AccountId',
       rentProjection: 'Option<RentProjection>'
     },
-    InstantiateReturnValue: {
-      _fallback: 'InstantiateReturnValueTo267',
+    InstantiateReturnValueOk: {
```
Comment (jacogr) [inline packages/types/src/interfaces/contracts/definitions.ts:277]: There is no previous type definition for this one, it is just replaced?

File: packages/types/src/interfaces/contracts/definitions.ts [hunk 2]
```diff
@@ -239,7 +245,7 @@ export default {
       value: 'Balance',
       gasLimit: 'Gas',
       storageDepositLimit: 'Option<Balance>',
-      code: 'Bytes',
+      code: 'CodeSource',
```
Comment (jacogr) [inline packages/types/src/interfaces/contracts/definitions.ts:248]: Also needs a previous version here.

**Review by jacogr — COMMENTED** [review]
Would just have older versions for the replaced types as well.

File: packages/types/src/interfaces/contracts/definitions.ts (hunk 2 above)
Comment (kwingram25) [reply packages/types/src/interfaces/contracts/definitions.ts:248]: I believe it was always supposed to be an enum, and Bytes was just a placeholder? Anyway, it seems like enum definitions can't have fallbacks at the moment, can they? I tried and it just made a struct with _enum and _fallback fields

**Comment by kwingram25** [comment]
Added fallback support for enums and sets. Any I missed?

**Review by jacogr — APPROVED** [review]
Brilliant!

**Pull request merged**